### PR TITLE
Show rejected ID cards

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Show cards for rejected identities.

## Changes

- Minor refactor of `IdCard` component

### CODE REVIEW

File: packages/browser-wallet/src/assets/svgX/rejected.svg
Comment (soerenbf) [inline packages/browser-wallet/src/assets/svgX/rejected.svg:1]: I'm sure this svg is already there with another colour. Can we just use that and apply `$color-red-attention` to it?

File: packages/browser-wallet/src/popup/popupX/pages/Onboarding/IdSubmitted.tsx [hunk 1]
```diff
@@ -21,7 +20,16 @@ export default function IdSubmitted() {
             <Page.Top heading={t('yourId')} />
             <Page.Main>
                 <Text.Capture>{t('idSubmitInfo')}</Text.Capture>
-                <IdCard rowsIdInfo={rowsIdInfo} idProviderName="TODO" identityName="TODO" />
+                <IdCard title="Identity 1" subtitle="Submitted to NotaBene">
```
Comment (soerenbf) [inline packages/browser-wallet/src/popup/popupX/pages/Onboarding/IdSubmitted.tsx:23]: Is this (i.e. "Identity 1", "Submitted by NotaBene") accurate?

File: packages/browser-wallet/src/popup/popupX/pages/Restore/RestoreResult.tsx [hunk 2]
```diff
@@ -30,7 +29,17 @@ export default function RestoreResult() {
             <Page.Top heading={t('result')} />
             <Page.Main>
                 <Text.Capture>{t('recoveredIds')}</Text.Capture>
-                <IdCard rowsIdInfo={rowsIdInfo} idProviderName="TODO" identityName="TODO" />
+
+                <IdCard title="Identity 1" subtitle="Verified by NotaBene">
```
Comment (soerenbf) [inline packages/browser-wallet/src/popup/popupX/pages/Restore/RestoreResult.tsx:33]: And here

File: packages/browser-wallet/src/popup/popupX/shared/IdCard/RejectedIdCard.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+export default function RejectedIdCard({ identity }: RejectedIdentityProps) {
+    const { t } = useTranslation('x', { keyPrefix: 'sharedX' });
+    const providers = useAtomValue(identityProvidersAtom);
+    const provider = providers.find((p) => p.ipInfo.ipIdentity === identity.providerIndex);
+    const idProviderName = provider?.ipInfo.ipDescription.name ?? 'Unknown';
+    return (
+        <IdCard title={identity.name} subtitle={t('idCard.rejectedBy', { idProviderName })}>
+            <IdCard.Content className="rejected">
+                <Rejected />
```
Comment (soerenbf) [inline packages/browser-wallet/src/popup/popupX/shared/IdCard/RejectedIdCard.tsx:23]: I don't know about the icon. It clearly states that there is an error, but I don't think it looks consistent when comparing with verified identities. IMO, removing the icon is better design-wise, but arguably less clear in terms of what's going on.

I would ask for design clarification here as well.

Personally, I would go for something like 
Screenshot 2024-11-13 at 09 45 51

But I don't think that looks fantastic either 🤷 

Also maybe it's redundant to state the rejection so many times as well? I think maybe remove it from the black part of the id card and just simply state the error detail received from the IDP?

File: packages/browser-wallet/src/popup/popupX/pages/Onboarding/IdSubmitted.tsx (hunk 1 above)
Comment (limemloh) [reply packages/browser-wallet/src/popup/popupX/pages/Onboarding/IdSubmitted.tsx:23]: These are again just placeholder text until we implement the onboarding flow.

File: packages/browser-wallet/src/popup/popupX/pages/Restore/RestoreResult.tsx (hunk 2 above)
Comment (limemloh) [reply packages/browser-wallet/src/popup/popupX/pages/Restore/RestoreResult.tsx:33]: lik above

File: packages/browser-wallet/src/popup/popupX/shared/IdCard/RejectedIdCard.tsx (hunk 3 above)
Comment (limemloh) [reply packages/browser-wallet/src/popup/popupX/shared/IdCard/RejectedIdCard.tsx:23]: I agree the design could be improved, but I would suggest we iterate on it in the design document and stick to this for now.

**Review by soerenbf — APPROVED** [review]

**Pull request merged**